course/models/models.py

```diff
@@ -456,6 +459,13 @@ def submit(self):
 class MultipleChoiceSubmission(Submission):
     @property
     def answer_display(self):
+        if self.uqj.is_checkbox():
+            values = []
+            for key, value in self.uqj.get_rendered_choices().items():
+                for answer in self.answer.split(','):
+                    if key == answer:
+                        values.append(value)
+            return values
         return self.uqj.get_rendered_choices().get(self.answer, 'Unknown')
```

Comment:
```suggestion
    @property
    def answer_display(self):
        values = []
        rendered_choices = self.uqj.get_rendered_choices()
        for answer in self.answer.split(','):
            values.append(rendered_choices.get(answer, 'Unknown'))
        return values  
```
We can't have a function that returns 2 different types of values.
Also, I'm not a fan of nested for loops in this case.
It also ignores the answers that are not in the choices. We should somehow specify something weird has happened.